src/pointer-analysis/value_set.cpp

```diff
@@ -1057,13 +1057,9 @@ void value_sett::get_reference_set_rec(
         insert(dest, exprt(ID_unknown, expr.type()));
       else
       {
-        index_exprt index_expr(expr.type());
-        index_expr.array()=object;
-        index_expr.index()=from_integer(0, index_type());
-
-        // adjust type?
-        if(ns.follow(object.type())!=array_type)
-          index_expr.make_typecast(array.type());
+        const index_exprt deref_index_expr(
+          typecast_exprt::conditional_cast(object, array_type),
```

Comment:
Note this is different to the previous code: by not following, we might create a cast to a synonym type

Reply:
Basically any test involving arrays which reach this part of the code, but I have not been able to construct any example where we would have an an object map that does _not_ have exactly one element.